### DISCUSSION TITLE
App: Condition Filter X and $CURRENT_ROLE _neq STRING

### DISCUSSION THREAD

**Discussion by CHSchuepfer** [directus/directus#15318] [Archive]
## Prelude

Hey everyone,

im currently stuck on a problem i cant seem to get my head around.
I have a dropdown which i wannt to deactivate if the Field "Stage" is "Production" and the current role of the User accessing this item isnt an admin.

Easy enough i thought, "Filter Condition" + Dynamic Variable "$CURRENT_ROLE" should take care of it.
Sadly, as it turns out im stuck for half a day and cant figure it out. Either im just having a really bad day or im missing something huge.

Can anyone help out and take a look please?

## Information:

- Directus: v9.16.1
- Deployed via: docker-compose

### v1:

```json
{
	"_and": [
		{
			"Stage": {
				"_eq": "Production"
			}
		},
		{
			"_and": []
		},
		{
			"$CURRENT_ROLE": {
				"_neq": "Administrator"
			}
		}
	]
}
```

### v2:
```json
{
	"_and": [
		{
			"Stage": {
				"_eq": "Production"
			}
		},
		{
			"_and": []
		},
		{
			"{{$CURRENT_ROLE}}": {
				"_neq": "Administrator"
			}
		}
	]
}
```

### v3:
```json
{
	"_and": [
		{
			"Stage": {
				"_eq": "Production"
			}
		},
		{
			"_and": []
		},
		{
			$CURRENT_ROLE: {
				"_neq": "Administrator"
			}
		}
	]
}
```

### v4:

```json
{
	"_and": [
		{
			"Stage": {
				"_eq": "Production"
			}
		},
		{
			"_and": []
		},
		{
			"$CURRENT_ROLE": {
				"name": {
					"_neq": "Administrator"
				}
			}
		}
	]
}
```

### v5:
```json
{
	"_and": [
		{
			"Stage": {
				"_eq": "Produktion"
			}
		},
		{
			"_and": [
				{
					"$CURRENT_ROLE": {
						"name": {
							"_neq": "Administrator"
						}
					}
				}
			]
		}
	]
}
```

I also checked the Name of group so it isnt something like "admin" and its a simple naming problem.

I also tried variations of v4&5 with $CURRENT_USER.role.id/name.
Sadly no luck. 

Can anyone see where the problem is?

Thanks in advance & so long :)

**Comment by rijkvanzanten** [comment]
Heya! Nested relational data isn't currently supported in Conditional Fields (see https://github.com/directus/directus/issues/8228). The value of `$CURRENT_ROLE` by default resolves to the `uuid` of the role, so you should still be able to achieve this by using something along the lines of:

```json
{
  "_and": [
    {
      "Stage": {
        "_eq": "Production"
      }
    },
    {
      "$CURRENT_ROLE": {
        "_eq": "e9dce429-409f-467c-8b56-390e78774c1d"
      }
    }
  ]
}
```

**Reply by Pachat** [reply]
Facing the same problem. I tried the proposed solution, but get a
```
{
  "errors": [
    {
      "message": "Cannot read properties of undefined (reading 'type')",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```
It looks like impossible to write `$CURRENT_ROLE` on the left side for Directus cannot guess its type.
So is there any other solution to apply the filter only if `$CURRENT_ROLE.admin_access` is `true`?

**Comment by CHSchuepfer** [comment]
Hey, thanks for the answer.

i forgot to mention i've already tried that in the whole "lets target the ID instead the name" thing.
So sadly no luck, but i will play around with it some more, maybe i come across a working solution.

In the end i can always block the actions triggered by the dropdown in my backend or even via flow.
So there is a quick and dirty way, i just like the idea of hiding the item in the first place. 
But i will tinker with it, some more :)


Thanks again and so long :)
